contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -1,87 +1,323 @@
 
-There is no supported way to change a feature gate at runtime for production Kubernetes use. A feature gate is typically toggled by a component restart.
+## Defining a feature gate
 
-Unless an exception is granted for a particular feature, as documented and approved as part of [Production Readiness Review], it is expected that:
+A feature gate definition is comprised of the following struct fields:
 
-- Toggling a feature gate will not affect other components, i.e. disabling a feature gate in the API server will work independently of disabling that same feature gate in the kubelet or scheduler, and no coordination between components is needed.
-- The effects of toggling a feature gate should be limited to the scope of the feature. Enabling or disabling a feature gate should not affect workloads that do not use the feature gate.
-- Toggling feature gates should not result in fanout effects or cascading interactions in a cluster.
-
-[Production Readiness Review]: /sig-architecture/production-readiness.md
-
-## Specification Fields
-	// Default is the default enablement state for the feature. Possible values are true/false.
+```
+	// Default is the default enablement state for the feature. If the value of
+	// the gate is not explicitly set, this value is used.
 	Default bool
 	
-	// LockToDefault indicates that the feature is locked to its default and cannot be changed. Possible values are true/false.
+	// LockToDefault indicates that the feature is locked to its default value
+	// and cannot be changed.
 	LockToDefault bool
 	
-	// PreRelease indicates the maturity level of the feature. possible values are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", "featuregate.Deprecated"
+	// PreRelease indicates the maturity level of the feature. Possible values
+	// are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", or
+	// "featuregate.Deprecated"
 	PreRelease prerelease
+```
+
+Unless an exception is granted for a particular feature, as documented and
+approved as part of [Production Readiness Review], it is expected that:
+
+- Toggling a feature gate will not affect other components, e.g. disabling a
+  feature gate in the API server will work independently of disabling that same
+  feature gate in the kubelet or scheduler, and no coordination between
+  components is needed.
+- The effects of toggling a feature gate should be limited to the scope of the
+  feature. Enabling or disabling a feature gate should not affect workloads
+  that do not use the feature gate.
+- Toggling feature gates should not result in fanout effects or cascading
+  interactions in a cluster.
+- Disabling a feature gate should prevent any further damage caused by bugs in
+  the feature.
+
+[Production Readiness Review]: /sig-architecture/production-readiness.md
+
+## Enabling and disabling feature gates
+
+Gates can be enabled or disabled either via flags (see `--feature-gates`) or
+component config files (see `featureGates`).
```

Comment:
heh, and since 1.30 as env variables for client-go https://github.com/kubernetes/kubernetes/pull/122555